### PR TITLE
Generate GMutex union with correct alignment constraints on 32 bit pl…

### PR DESCRIPTION
**Pull request by sdroege**
…atforms

While it is always 64 bits in size, on 32 bit platforms it only requires
32 bit alignment due to being expressed by 2x32 bit integers.

### CODE REVIEW

File: src/codegen/sys/lib_.rs [hunk 1]
```diff
@@ -256,7 +256,8 @@ fn generate_unions(w: &mut Write, env: &Env, items: &[&Union])
             if env.config.library_name == "GLib" && c_type == "GMutex" {
                 // Two c_uint or a pointer => 64 bits on all
                 // platforms currently supported by GLib
-                try!(writeln!(w, "#[repr(C)]\npub struct {}(u64); // union", c_type));
+                try!(writeln!(w, "#[cfg(target_pointer_width = \"32\")]\n#[repr(C)]\npub struct {0}([u32; 2]);\n\
+                                  #[cfg(target_pointer_width = \"64\")]\n#[repr(C)]\npub struct {0}(*mut c_void);", c_type));
```
Comment (EPashkin) [inline src/codegen/sys/lib_.rs]: Why not leave 64bit version u64?

Comment (EPashkin) [inline src/codegen/sys/lib_.rs]: And IMHO need updated comment about alignment

Comment (sdroege) [inline src/codegen/sys/lib_.rs]: It's a pointer in C, so I thought it's cleaner to make it a pointer in Rust too

Comment (EPashkin) [inline src/codegen/sys/lib_.rs]: Ok, only comment last

Comment (sdroege) [inline src/codegen/sys/lib_.rs]: Ok, done :)

**Comment by EPashkin** [comment]
Thanks

**Pull request merged**